service/src/main/kotlin/fi/espoo/evaka/invoicing/controller/VoucherValueDecisionController.kt

```diff
@@ -52,6 +82,81 @@ class VoucherValueDecisionController(private val jdbi: Jdbi) {
             else (total / pageSize) + 1
         return ResponseEntity.ok(VoucherValueDecisionSearchResult(valueDecisions, total, pages))
     }
+
+    @GetMapping("/{id}")
+    fun getDecision(
+        user: AuthenticatedUser,
+        @PathVariable id: UUID
+    ): ResponseEntity<Wrapper<VoucherValueDecisionDetailed>> {
+        Audit.VoucherValueDecisionRead.log(targetId = id)
+        user.requireOneOfRoles(Roles.FINANCE_ADMIN)
+        val res = jdbi.transaction { it.getVoucherValueDecision(objectMapper, id) }
+            ?: throw NotFound("No voucher value decision found with given ID ($id)")
+        return ResponseEntity.ok(Wrapper(res))
+    }
+
+    @PostMapping("/send")
+    fun sendDrafts(user: AuthenticatedUser, @RequestBody decisionIds: List<UUID>): ResponseEntity<Unit> {
+        Audit.VoucherValueDecisionSend.log(targetId = decisionIds)
+        user.requireOneOfRoles(Roles.FINANCE_ADMIN)
+        jdbi.transaction { sendVoucherValueDecisions(it, user, decisionIds) }
+        asyncJobRunner.scheduleImmediateRun()
+        return ResponseEntity.noContent().build()
+    }
+
+    @GetMapping("/pdf/{id}")
+    fun getDecisionPdf(user: AuthenticatedUser, @PathVariable id: UUID): ResponseEntity<ByteArray> {
+        Audit.FeeDecisionPdfRead.log(targetId = id)
+        user.requireOneOfRoles(Roles.FINANCE_ADMIN)
+        val (filename, pdf) = jdbi.transaction { h -> valueDecisionService.getDecisionPdf(h, id) }
+        val headers = HttpHeaders().apply {
+            add("Content-Disposition", "attachment; filename=\"$filename\"")
+            add("Content-Type", "application/pdf")
+        }
+        return ResponseEntity(pdf, headers, HttpStatus.OK)
+    }
+
+    private fun sendVoucherValueDecisions(h: Handle, user: AuthenticatedUser, ids: List<UUID>) {
+        h.lockValueDecisions(ids)
+        val decisions = h.getValueDecisionsByIds(objectMapper, ids)
+        if (decisions.isEmpty()) return
+
+        val notDrafts = decisions.filterNot { it.status == VoucherValueDecisionStatus.DRAFT }
```

Comment:
Personal preference: I think `any`/`all` are more readable than filtering the actual list elements and checking the resulting length:

```
if (decisions.any { it.status != VoucherValueDecisionStatus.DRAFT }) {
  throw BadRequest("Some voucher value decisions were not drafts")
}
```

error message matches the check exactly